### PR TITLE
Test cases for Package_Item(Model)

### PR DESCRIPTION
**Pull request by Jeshan7**
## Why?
    - Coverage of package_item tests were low 

## This change addresses the need by:
   - Updated the test cases in test/data/model/package_item_test.exs
   - Wrote new test cases in test/data/model/package_item_test.exs

[delivers #163248592]

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

### CODE REVIEW

File: apps/snitch_core/test/data/model/package_item_test.exs [hunk 1]
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
...
-                   package_id: -1
-               })
-
-      refute changeset.valid?
-      assert %{package_id: ["does not exist"]} == errors_on(changeset)
-
-      assert {:error, changeset} =
+    test "successful with valid attributes", context do
+      %{package_item: package_item} = new_package(context)
+      assert {:ok, _} =
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/package_item_test.exs:34]: The approach here is not correct.

You only have to create an empty package, then use the line_item you have set up to create a package item. You are already creating a package_item in `new_package` and using the attributes of that to create another package_item.

File: apps/snitch_core/test/data/model/package_item_test.exs [hunk 2]
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
...
-      %{line_items: [line_item]} = context
-
-      assert {:ok, package_item} =
-               PackageItem.create(%{
-                 @params
-                 | line_item_id: line_item.id,
-                   product_id: line_item.product_id,
-                   package_id: make_package(context).id
-               })
+    test "with invalid attributes" do
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/model/package_item_test.exs]: Test statement is incorrect

File: apps/snitch_core/test/data/model/package_item_test.exs [hunk 3]
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
...
+      params = %{quantity: 95}
+      {:ok, updated_package_item} = PackageItem.update(params, package_item)
+      assert updated_package_item.id == package_item.id
+      assert updated_package_item.quantity != package_item.quantity
+      refute updated_package_item.backordered?
+    end
 
-    test "with valid params", context do
-      %{line_items: [line_item]} = context
+    test "with invalid params", context do
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/model/package_item_test.exs]: Test statement is incorrect

File: apps/snitch_core/test/data/model/package_item_test.exs [hunk 4]
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
...
-    } = context
-
-    insert(
-      :package,
-      order: order,
-      shipping_category: sc,
-      shipping_method_id: sm.id
-    )
+
+  defp new_package(context) do
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/package_item_test.exs:110]: Break the function into two one should create a `package` another should create `package_item` and you should pass params which are required explicitly. You can make use of the context. The way `embedded_shipping_methods` is passed.

File: apps/snitch_core/test/data/model/package_item_test.exs [hunk 5]
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
...
-                 %{
-                   state: "destroyed",
-                   quantity: 5,
-                   delta: 0
-                 },
-                 package_item
-               )
+    test "with a map", context do
+      %{package_item: package_item} = new_package(context)
+      map = %{quantity: package_item.quantity, delta: package_item.delta}
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/package_item_test.exs]: Use something which is unique for `package item`  e.g. package number. The keys that you have used are not unique for a package item. You have used quantity and delta, do you think these are unique for `package items`?

**Review by arjun289 — CHANGES_REQUESTED** [review]

**Review by SagarKarwande — CHANGES_REQUESTED** [review]

**Review by arjun289 — APPROVED** [review]
Looks good now.

**Review by SagarKarwande — APPROVED** [review]

**Pull request merged**